test/wallet-test.js

```diff
@@ -1912,6 +1913,54 @@ describe('Wallet', function() {
     }
   });
 
+  it('should not exceed MEMPOOL_MAX_ANCESTORS policy', async () => {
+    // Create wallet and get one address
+    const wallet = await wdb.create();
+    const addr1 = await wallet.receiveAddress();
+
+    // Dummy address for outputs
+    const recAddr = Address.fromHash(Buffer.alloc(20, 1));
+
+    // Add one single, unconfirmed coin to wallet
+    const mtx1 = new MTX();
+    mtx1.addInput(dummyInput());
+    mtx1.addOutput(addr1, 10 * 1e8);
+    const tx1 = mtx1.toTX();
+    await wallet.txdb.add(tx1, null);
+
+    // Spend unconfirmed change outputs up to the limit
+    const limit = policy.MEMPOOL_MAX_ANCESTORS;
+    for (let i = 0; i < limit; i++) {
+      const tx = await wallet.send({
+        outputs: [{
+          address: recAddr,
+          value: 10000}]
+      });
+      assert(await wallet.txdb.hasPending(tx.hash()));
+    }
+
+    // At the limit
+    const pending = await wallet.getPending();
+    assert(pending.length = policy.MEMPOOL_MAX_ANCESTORS);
```

Comment:
This is object assignment, not equality check

Reply:
oy, thank you. 💀

Reply:
This mistake let an off-by-one error through.... fixing!